pynvim/api/common.py

```diff
@@ -85,7 +85,10 @@ def __init__(self, obj, get_method, set_method=None):
 
     def __getitem__(self, key):
         """Return a map value by key."""
-        return self._get(key)
+        try:
+            return self._get(key)
+        except Exception:
+            raise KeyError
```

Comment:
Is the generic `KeyError` the proper thing to throw for the API?

Maybe throw `NvimAPIKeyError` which derives from `KeyError`? Would be more precise and `exception KeyError` in user code would still work.

Reply:
Had the same discussion in https://github.com/neovim/pynvim/pull/405#discussion_r315202545 already.